File: codeforces/problems/34B_Sale/main.go
package main

import (
	"fmt"
	"sort"
)

func main() {
	var n, m int
	fmt.Scan(&n, &m)

	price := make([]int, n)
	for i := 0; i < n; i++ {
		fmt.Scan(&price[i])
	}

	sort.Ints(price)

	result := 0
	count := 0

	for i := 0; i < n && count < m; i++ {
		if price[i] < 0 {
			result += -price[i]
			count++
		}
	}

	fmt.Println(result)
}

File: codeforces/problems/34B_Sale/main.cpp
#include <iostream>
#include <vector>
#include <algorithm>

int main() {
    int n, m;
    std::cin >> n >> m;

    std::vector<int> prices(n);
    for (int i = 0; i < n; ++i) {
        std::cin >> prices[i];
    }

    std::sort(prices.begin(), prices.end());

    int result = 0;
    int count = 0;

    for (int i = 0; i < n && count < m; ++i) {
        if (prices[i] < 0) {
            result += -prices[i];
            ++count;
        }
    }

    std::cout << result << std::endl;

    return 0;
}
